BigQuery connection error on 0.41.1

**Describe the bug**
I've updated Metabase from **0.40.4** to **0.41.1** and the old BigQuery connection has broken.
When I try to use new driver I see `com.google.api.client.http.HttpTransport.isMtls()Z` error.
When I try to use the old driver I see `No method in multimethod 'can-connect?' for dispatch value: :bigquery`

Also, I've tried to issue a new service key and tested this key via python-bigquery — it works ok. Both service accounts has BigQuery admin permissions. I've tried to set up new database — it's not working.

**Logs**
```
[8e1ca244-512d-4fae-bf99-ba68c9167ad4] 2021-11-01T21:45:40+03:00 ERROR metabase.driver.util Database connection error
java.lang.NoSuchMethodError: com.google.api.client.http.HttpTransport.isMtls()Z
	at com.google.api.services.bigquery.Bigquery$Builder.chooseEndpoint(Bigquery.java:6381)
	at com.google.api.services.bigquery.Bigquery$Builder.<init>(Bigquery.java:6414)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.<init>(HttpBigQueryRpc.java:107)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:56)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:50)
	at com.google.cloud.ServiceOptions.getRpc(ServiceOptions.java:560)
	at com.google.cloud.bigquery.BigQueryOptions.getBigQueryRpcV2(BigQueryOptions.java:125)
	at com.google.cloud.bigquery.BigQueryImpl.<init>(BigQueryImpl.java:244)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:46)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:40)
	at com.google.cloud.ServiceOptions.getService(ServiceOptions.java:540)
	at metabase.driver.bigquery_cloud_sdk$database__GT_client.invokeStatic(bigquery_cloud_sdk.clj:51)
	at metabase.driver.bigquery_cloud_sdk$database__GT_client.invoke(bigquery_cloud_sdk.clj:46)
	at metabase.driver.bigquery_cloud_sdk$fn__71435.invokeStatic(bigquery_cloud_sdk.clj:76)
	at metabase.driver.bigquery_cloud_sdk$fn__71435.invoke(bigquery_cloud_sdk.clj:73)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__24606.invoke(util.clj:43)
	at metabase.util$do_with_timeout$fn__6173.invoke(util.clj:354)
	at clojure.core$binding_conveyor_fn$fn__5772.invoke(core.clj:2034)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[8e1ca244-512d-4fae-bf99-ba68c9167ad4] 2021-11-01T21:45:40+03:00 ERROR metabase.api.database Cannot connect to Database
java.lang.Exception: com.google.api.client.http.HttpTransport.isMtls()Z
	at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:48)
	at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:31)
	at clojure.lang.RestFn.invoke(RestFn.java:442)
	at metabase.api.database$test_database_connection.invokeStatic(database.clj:414)
	at metabase.api.database$test_database_connection.doInvoke(database.clj:404)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.api.database$fn__77102.invokeStatic(database.clj:561)
	at metabase.api.database$fn__77102.invoke(database.clj:541)
	at compojure.core$wrap_response$fn__32043.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__32027.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__32032.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__32036.invoke(core.clj:151)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32036.invoke(core.clj:153)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__70241.invoke(auth.clj:14)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055.invoke(core.clj:200)
	at compojure.core$make_context$handler__32083.invoke(core.clj:289)
	at compojure.core$make_context$fn__32087.invoke(core.clj:299)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$make_context$fn__32087.invoke(core.clj:300)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at metabase.api.routes$fn__79716$fn__79719.invoke(routes.clj:56)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.server.routes$fn__79861$fn__79862.doInvoke(routes.clj:57)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055.invoke(core.clj:200)
	at compojure.core$make_context$handler__32083.invoke(core.clj:289)
	at compojure.core$make_context$fn__32087.invoke(core.clj:299)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32036.invoke(core.clj:153)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32036.invoke(core.clj:153)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32036.invoke(core.clj:153)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055$f__32056$respond_SINGLEQUOTE___32057.invoke(core.clj:197)
	at metabase.server.routes$fn__79849$fn__79851.invoke(routes.clj:41)
	at compojure.core$routes$fn__32055$f__32056.invoke(core.clj:198)
	at compojure.core$routes$fn__32055.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__77323.invoke(exceptions.clj:98)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__77320.invoke(exceptions.clj:86)
	at metabase.server.middleware.log$log_api_call$fn__80316$fn__80317$fn__80318.invoke(log.clj:209)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invokeStatic(diagnostic.clj:15)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invoke(diagnostic.clj:9)
	at metabase.server.middleware.log$log_api_call$fn__80316$fn__80317.invoke(log.clj:201)
	at toucan.db$_do_with_call_counting.invokeStatic(db.clj:216)
	at toucan.db$_do_with_call_counting.invoke(db.clj:209)
	at metabase.server.middleware.log$log_api_call$fn__80316.invoke(log.clj:200)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__83529.invoke(browser_cookie.clj:30)
	at metabase.server.middleware.security$add_security_headers$fn__59186.invoke(security.clj:144)
	at metabase.server.middleware.json$wrap_json_body$fn__82673.invoke(json.clj:62)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__82691.invoke(json.clj:98)
	at metabase.server.middleware.offset_paging$handle_paging$fn__59210.invoke(offset_paging.clj:42)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__83796.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__83812.invoke(params.clj:69)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__34054.invoke(misc.clj:59)
	at metabase.server.middleware.session$bind_current_user$fn__44955$fn__44956.invoke(session.clj:257)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:238)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:230)
	at metabase.server.middleware.session$bind_current_user$fn__44955.invoke(session.clj:256)
	at metabase.server.middleware.session$wrap_current_user_info$fn__44942.invoke(session.clj:216)
	at metabase.server.middleware.session$wrap_session_id$fn__44928.invoke(session.clj:162)
	at metabase.server.middleware.auth$wrap_api_key$fn__70249.invoke(auth.clj:27)
	at ring.middleware.cookies$wrap_cookies$fn__83716.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__34037.invoke(misc.clj:27)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__34062.invoke(misc.clj:76)
	at ring.middleware.gzip$wrap_gzip$fn__83758.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__34065.invoke(misc.clj:93)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__83545.invoke(ssl.clj:48)
	at metabase.server$async_proxy_handler$fn__80089.invoke(server.clj:71)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:386)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoSuchMethodError: com.google.api.client.http.HttpTransport.isMtls()Z
	at com.google.api.services.bigquery.Bigquery$Builder.chooseEndpoint(Bigquery.java:6381)
	at com.google.api.services.bigquery.Bigquery$Builder.<init>(Bigquery.java:6414)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.<init>(HttpBigQueryRpc.java:107)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:56)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryRpcFactory.create(BigQueryOptions.java:50)
	at com.google.cloud.ServiceOptions.getRpc(ServiceOptions.java:560)
	at com.google.cloud.bigquery.BigQueryOptions.getBigQueryRpcV2(BigQueryOptions.java:125)
	at com.google.cloud.bigquery.BigQueryImpl.<init>(BigQueryImpl.java:244)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:46)
	at com.google.cloud.bigquery.BigQueryOptions$DefaultBigQueryFactory.create(BigQueryOptions.java:40)
	at com.google.cloud.ServiceOptions.getService(ServiceOptions.java:540)
	at metabase.driver.bigquery_cloud_sdk$database__GT_client.invokeStatic(bigquery_cloud_sdk.clj:51)
	at metabase.driver.bigquery_cloud_sdk$database__GT_client.invoke(bigquery_cloud_sdk.clj:46)
	at metabase.driver.bigquery_cloud_sdk$fn__71435.invokeStatic(bigquery_cloud_sdk.clj:76)
	at metabase.driver.bigquery_cloud_sdk$fn__71435.invoke(bigquery_cloud_sdk.clj:73)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__24606.invoke(util.clj:43)
	at metabase.util$do_with_timeout$fn__6173.invoke(util.clj:354)
	at clojure.core$binding_conveyor_fn$fn__5772.invoke(core.clj:2034)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
[8e1ca244-512d-4fae-bf99-ba68c9167ad4] 2021-11-01T21:45:40+03:00 DEBUG metabase.server.middleware.log PUT /api/database/4 400 61.7 ms (1 DB calls) 
{:valid false,
 :dbname "com.google.api.client.http.HttpTransport.isMtls()Z",
 :message "com.google.api.client.http.HttpTransport.isMtls()Z"}
```

**To Reproduce**
Steps to reproduce the behavior:
1. Add new database
2. Make new service key with BigQuery Admin permissions

**Expected behavior**
It should work. 

**Screenshots**
<img width="818" alt="image" src="https://user-images.githubusercontent.com/40689054/139725120-d996717b-0935-496c-86a6-853163eb4c6a.png">
<img width="816" alt="image" src="https://user-images.githubusercontent.com/40689054/139725162-dbccf9c0-d64f-4430-9fc4-5f692b452332.png">
<img width="600" alt="image" src="https://user-images.githubusercontent.com/40689054/139725137-662ed243-bacd-4d90-a2a4-8d51f538bda4.png">



**Severity**
Extremely severe — we have no access to our data from Metabase ¯\_( ツ)_/¯ 


**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.1 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "1.8.0_302-8u302-b08-1~deb9u1-b08",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "1.8.0_302",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "25.302-b08",
    "os.name": "Linux",
    "os.version": "5.4.144+",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "googleanalytics",
      "postgres",
      "snowflake",
      "bigquery-cloud-sdk"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.4"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.23"
      }
    },
    "run-mode": "prod",
    "version": {
      "tag": "v0.41.1",
      "date": "2021-10-21",
      "branch": "release-x.41.x",
      "hash": "76aa4a5"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

--- a/modules/drivers/bigquery/deps.edn
+++ b/modules/drivers/bigquery/deps.edn
@@ -3,8 +3,8 @@
 
  :deps
  ;; TODO: use BOM to manage these (see extensive comment in project.clj under bigquery for more details)
- {com.google.apis/google-api-services-bigquery {:mvn/version "v2-rev20200523-1.30.9"}
-  ;; keep this synced with bigquery-cloud-sdk driver dependency version
-  com.google.http-client/google-http-client-jackson2 {:mvn/version "1.39.2"}}
+ ;; ensure the version of google-http-client transitively depended on here matches that of bigquery-cloud-sdk
+ ;; (which, at the moment, is 1.39.2)
+ {com.google.apis/google-api-services-bigquery {:mvn/version "v2-rev20211106-1.32.1"}}
 
  :metabase.driver/parents #{:google}}

--- a/modules/drivers/googleanalytics/deps.edn
+++ b/modules/drivers/googleanalytics/deps.edn
@@ -1,6 +1,8 @@
 {:paths ["src" "resources"]
 
  :deps
- {com.google.apis/google-api-services-analytics {:mvn/version "v3-rev20180622-1.27.0"}}
+ ;; ensure the version of google-http-client transitively depended on here matches that of bigquery-cloud-sdk
+ ;; (which, at the moment, is 1.39.2)
+ {com.google.apis/google-api-services-analytics {:mvn/version "v3-rev20190807-1.32.1"}}
 
  :metabase.driver/parents #{:google}}

